ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -239,6 +239,7 @@ public static ScheduledExecutorService newScheduler(int numThreads, String threa
         new ScheduledThreadPoolExecutor(numThreads, new SchedulerThreadFactory(threadNamePrefix, isDaemon));
     scheduler.setContinueExistingPeriodicTasksAfterShutdownPolicy(false);
     scheduler.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
+    scheduler.setRemoveOnCancelPolicy(true);
```

Comment:
I can remove it for now. Since we are not constantly rescheduling jobs, there is no risk of memory leak by leaving it to false.